File: jour03/p3.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <map>
using namespace std;

int main() {
    fstream monFichier("input.txt", ios::in);

    string s;
    int somme = 0;

    while (getline(monFichier, s)) {
        size_t indice = s.find("mul(");
        while (indice !=std::string::npos) {
            indice += 4;
            int n1 = 0;
            int i= 0;

            while (s[indice + i] >= '0' && s[indice + i] <= '9') {
                n1 = n1 * 10 + (s[indice + i] - '0');
                i++;
            }
            
            if (s[indice + i] == ',' && n1 < 1000) {
                i += 1;
                int n2 = 0;
                while (s[indice + i] >= '0' && s[indice + i] <= '9') {
                    n2 = n2 * 10 + (s[indice + i] - '0');
                    i += 1;
                }

                if (s[indice + i] == ')' && n2 < 1000) {
                    somme += n1 * n2;
                }

            }

            s = s.substr(indice);
            indice = s.find("mul(");
        }
    }

    cout << somme << endl;
}

File: jour03/p3.py
fic = open("input.txt", "r")

somme = 0

for ligne in fic:
    indice = ligne.find("mul(")
    while indice != -1:
        indice = indice + 4
        n1 = 0
        taille = 0
        while '0' <= ligne[indice + taille] <= '9':
            n1 = n1 * 10 + int(ligne[indice + taille])
            taille += 1

        if (ligne[indice + taille] == "," and n1 < 1000):
            taille += 1
            n2 = 0
            while '0' <= ligne[indice + taille] <= '9':
                n2 = n2 * 10 + int(ligne[indice + taille])
                taille += 1

            if (ligne[indice + taille] == ")" and n2 < 1000):
                somme += n1 * n2
        ligne = ligne[indice + taille:]
        indice = ligne.find("mul(")
print(somme)